### PR TITLE
add guide on how to add a separate website maintained by Sourcegraph

### PR DESCRIPTION
**Pull request by vanesa**
Documentation on how to create a separate website maintained by Sourcegraph, e.g. [LSIF.dev](https://lsif.dev) or [langserver.org](https://langserver.org).

### CODE REVIEW

File: doc/dev/documentation/separate_website.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# How to setup a separate website maintained by Sourcegraph
+
+    💡 Use this template to describe the steps engineers should follow to setup a separate website maintained by Sourcegraph. 
+
+# 1. When do we need a separate website?
+
+For projects like [langserver.org](http://langserver.org) or [lsif.dev](http://lsif.dev), we prefer to create separate websites from 
+[Sourcegraph.com](http://sourcegraph.com). In these two examples, we provide a list of LSPs or LSIF indexers offered by us or other parties 
+that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tide to Sourcegraph and we are not
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tied to Sourcegraph and we are not 
```

File: doc/dev/documentation/separate_website.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    💡 Use this template to describe the steps engineers should follow to setup a separate website maintained by Sourcegraph. 
+
+# 1. When do we need a separate website?
+
+For projects like [langserver.org](http://langserver.org) or [lsif.dev](http://lsif.dev), we prefer to create separate websites from 
+[Sourcegraph.com](http://sourcegraph.com). In these two examples, we provide a list of LSPs or LSIF indexers offered by us or other parties 
+that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tide to Sourcegraph and we are not 
+the sole providers, we publish this information on an independent website.
+
+# 2. Setup domain and create content
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
# 2. Set up domain and create content
```
I believe we use two words when used as a verb, one word when used as a noun

File: doc/dev/documentation/separate_website.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        [project url]
+
+    - README.md
+    - index.html
+    - style-addition.css
+
+When all the content is created, we `push` our branch, create a **PR** and ask for reviews and merge to master.
+
+Examples: [Langserver](http://github.com/langserver), [LSIF](http://github.com/lsif)
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]: These appear to be links to the orgs, not the repos

File: doc/dev/documentation/separate_website.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+# 3. Update the DNS to point to the server
+
+We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.
+
+If you don't have terraform installed and install via ```brew install terraform```:
+
+    💡 Ensure that your local `terrraform` version is the same as the `CI` version to **avoid file locks**.
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
    💡 Ensure that your local `terraform` version is the same as the `CI` version to **avoid file locks**.
```

File: doc/dev/documentation/separate_website.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - index.html
+    - style-addition.css
+
+When all the content is created, we `push` our branch, create a **PR** and ask for reviews and merge to master.
+
+Examples: [Langserver](http://github.com/langserver), [LSIF](http://github.com/lsif)
+
+# 3. Update the DNS to point to the server
+
+We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to a non-`master` branch and get it reviewed before merging.
```
Ignore this if I'm misinterpreting this one

File: doc/dev/documentation/separate_website.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+If you don't have terraform installed and install via ```brew install terraform```:
+
+    💡 Ensure that your local `terrraform` version is the same as the `CI` version to **avoid file locks**.
+
+    💡 Make sure to pull the **latest version** of the infrastructure `master` before creating your branch.
+
+    💡 Check that all of the domain's `A records` are also referenced in the terraform file.
+
+Inside your branch:
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
From the `infrastructure` repository, with your branch checked out:
```

File: doc/dev/documentation/separate_website.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+First step is to request @beyang or @sqs get a domain and add the site to Sourcegraph's Cloudflare account. 
+    
+    💡 Make sure `www` is redirecting to `https`.
+
+### Create the website content
+
+Once the domain is secured, we need to setup the GitHub page, where the static page will be hosted as: `[projectname].github.io`.
+
+We clone the repo locally, create a branch from master (make sure master is up-to-date!) and add the following files:
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
We clone the repository locally, create a branch from master (make sure master is up-to-date!) and add the following files:
```

File: doc/dev/documentation/separate_website.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+Inside your branch:
+
+    $ terraform apply
+    $ git commit -m 'dns: Updated terraform.tfstate' terraform.tfstate
+    $ git push origin branch
+
+Get the branch reviewed and merge to `master`.
+
+
+Other Sourcegraph docs:
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]: Hah — this is new. Is the idea just to list random docs, or are these somehow related to this particular one?

File: doc/dev/documentation/separate_website.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+For projects like [langserver.org](http://langserver.org) or [lsif.dev](http://lsif.dev), we prefer to create separate websites from 
+[Sourcegraph.com](http://sourcegraph.com). In these two examples, we provide a list of LSPs or LSIF indexers offered by us or other parties 
+that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tide to Sourcegraph and we are not 
+the sole providers, we publish this information on an independent website.
+
+# 2. Setup domain and create content
+
+### Secure the domain
+
+First step is to request @beyang or @sqs get a domain and add the site to Sourcegraph's Cloudflare account.
```
Comment (dadlerj) [inline doc/dev/documentation/separate_website.md]:
```suggestion
First step is to request @beyang get a domain and add the site to Sourcegraph's Cloudflare account. 
```

File: doc/dev/documentation/separate_website.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.
+
+If you don't have terraform installed and install via ```brew install terraform```:
+
+    💡 Ensure that your local `terrraform` version is the same as the `CI` version to **avoid file locks**.
+
+    💡 Make sure to pull the **latest version** of the infrastructure `master` before creating your branch.
+
+    💡 Check that all of the domain's `A records` are also referenced in the terraform file.
```
Comment (nicksnyder) [inline doc/dev/documentation/separate_website.md]: The formatting here seems a bit unusual. Can this be an ordered/unordered list?

File: doc/dev/documentation/separate_website.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (vanesa) [reply doc/dev/documentation/separate_website.md]: The style guide for documentation states to use relative paths. Should I make an exemption here since it's so far out?

File: doc/dev/documentation/separate_website.md (hunk 8 above)
Comment (vanesa) [reply doc/dev/documentation/separate_website.md]: Yes, sorry, I thought that this was the reason why the build keeps failing as I get this error:
```
dev/documentation/separate_website.md: disconnected page (no inlinks from other pages)
--
  | [2019-10-15T01:07:37Z] 1 problems found

```
So I thought maybe I need to mention other doc links for it to pass, but I guess that doesn't solve it  😅. I can remove it again. Any idea what the build error means?

File: doc/dev/documentation/separate_website.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicksnyder) [reply doc/dev/documentation/separate_website.md]: Relative paths only make sense within the sourcegraph/sourcegraph repo. This link will break if I try to click on this link locally and don't have sourcegraph/infrastructure checked out as a sibling. Can you update the docs that you are referring to to clarify this?

File: doc/dev/documentation/separate_website.md (hunk 8 above)
Comment (nicksnyder) [reply doc/dev/documentation/separate_website.md]: You need to link *to* this page from somewhere (e.g. from index.md)
